src/liballoc/sync.rs

```diff
@@ -751,6 +751,59 @@ impl<T: ?Sized> Arc<T> {
         this.inner().strong.load(SeqCst)
     }
 
+    /// Increments the strong reference count on the `Arc<T>` associated with the
+    /// provided pointer by one.
+    ///
+    /// # Safety
+    ///
+    /// The pointer must have been obtained through `Arc::into_raw`, and the
+    /// associated `Arc` instance must be valid (i.e. the strong count must be at
+    /// least 1) for the duration of this method.
+    #[inline]
+    #[unstable(feature = "arc_mutate_strong_count", issue = "none")]
+    pub unsafe fn incr_strong_count(ptr: *const T) {
+        let this = Arc::from_raw(ptr);
```

Comment:
I think this function is currently a no-op, since the reference count is decremented again when `this` is dropped.

Reply:
Oh excellent catch; you're right!